protocols/mdns/src/behaviour.rs

```diff
@@ -115,29 +126,54 @@ pub struct Mdns {
 
     /// Discovery timer.
     timeout: Timer,
+
+    /// IP Type.
+    domain: IPType,
 }
 
 impl Mdns {
     /// Builds a new `Mdns` behaviour.
     pub async fn new(config: MdnsConfig) -> io::Result<Self> {
-        let recv_socket = {
-            let socket = Socket::new(
-                Domain::ipv4(),
-                Type::dgram(),
-                Some(socket2::Protocol::udp()),
-            )?;
-            socket.set_reuse_address(true)?;
-            #[cfg(unix)]
-            socket.set_reuse_port(true)?;
-            socket.bind(&SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 5353).into())?;
-            let socket = socket.into_udp_socket();
-            socket.set_multicast_loop_v4(true)?;
-            socket.set_multicast_ttl_v4(255)?;
-            Async::new(socket)?
+        let recv_socket = match config.domain {
+            IPType::IPv4 => {
+                let socket = Socket::new(
+                    Domain::ipv4(),
+                    Type::dgram(),
+                    Some(socket2::Protocol::udp()),
+                )?;
+                socket.set_reuse_address(true)?;
+                #[cfg(unix)]
+                socket.set_reuse_port(true)?;
+                socket.bind(&SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 5353).into())?;
+                let socket = socket.into_udp_socket();
+                socket.set_multicast_loop_v4(true)?;
+                socket.set_multicast_ttl_v4(255)?;
+                Async::new(socket)?
```

Comment:
I think you can remove the v4 socket. Unless the `IPV6_V6ONLY` socket option is enabled a v6 socket should support both ip protocols.

Reply:
so this is actually needed?

Reply:
it is used in line 397, might be created there but wanted avoid not needed object creation

Reply:
and yes this is not needed in case of Linux, but in that case mdns will discover also IPv6 addresses